hedy.py

```diff
@@ -665,16 +665,25 @@ def program(self, args):
                     return False, a[1]
 
     #leafs are treated differently, they are True + their arguments flattened
-    def var(self, args):
-        return True, ''.join([str(c) for c in args])
-    def random(self, args):
-        return True, 'random'
-    def punctuation(self, args):
-        return True, ''.join([c for c in args])
-    def number(self, args):
-        return True, ''.join([c for c in args])
-    def text(self, args):
-        return all(args), ''.join([c for c in args])
+    @v_args(meta=True)
```

Comment:
Ah I see why this means we only have access to the leafs. Could we add a default rule here that passes `meta`?

Reply:
(and then calls the method on the inheriting method)